core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2909,16 +2894,31 @@ struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
         Kokkos::Tools::Impl::begin_parallel_for(policy, *this, functor_name,
```

Comment:
I'm lukewarm about calling the full begin_parallel_for version. I'll think more about it, but now begin_parallel_for (rather than beginParallelFor) implies tuning, and could mean more. I'm leaning towards wanting beginParallelFor calls here, though there is a good reason to do what you've done

Reply:
(that good reason being that we don't have multiple codepaths to achieve the same effect)

Reply:
I changed to `beginParallelFor` now after rebasing. The code duplication for the profiling is fairly small.

For the non-scalar version, we were always using `Kokkos::Tools::Impl::begin_parallel_for` while we were using `Kokkos::Profiling::beginParallelFor` for the scalar version. Is that difference intentional?

Reply:
No, that difference was not intentional.